tests/io/test_defaultnode.py

```diff
@@ -46,3 +57,31 @@ def test_md5(unode, xfs):
         "/node/dir/file2", contents="The quick brown fox jumps over the lazy dog"
     )
     assert unode.io.md5("dir/file2") == "9e107d9d372bb6826bd81d3542a419d6"
+
+
+def test_reserve_bytes(unode, xfs):
+    """test byte reservations"""
+
+    xfs.create_dir("/node")
+    xfs.set_disk_usage(10000)
+
+    # We start off with 10,000 bytes available
+    assert unode.io.reserve_bytes(4000, check_only=True) is True
+    assert unode.io.reserve_bytes(40000) is False
+
+    # Now reserve some bytes
+    assert unode.io.reserve_bytes(30000) is False
+
+    assert unode.io.reserve_bytes(2000) is True  # 4k bytes reserved
+    assert unode.io.reserve_bytes(2000) is True  # 8k bytes reserved
+    assert unode.io.reserve_bytes(2000) is False  # can't reserve 12k bytes
```

Comment:
I'll make the comments more explicit.

There's a factor of two when calling `reserve_bytes`:
https://github.com/radiocosmology/alpenhorn/blob/c4ead5d57b3ebcb7d0a5b32e03c1083b2aa4ee9d/alpenhorn/io/default.py#L264
which is present to avoid having to figure out how much space on disk a file of a given size is actually going to use up.  In this test, we start with 10,000 bytes free/unreserved.
* The first call reserves 4000 bytes (2 x 2000), leaving 6000 unreserved.
* The second call reserves another 4000 bytes (2 x 2000), leaving 2000 unreserved.
* The third call wants to reserve 4000 bytes again, but as there's only 2000 left unreserved at this point, it fails.

The factor of two is a _very_ conservative number, but I don't it matters too much: a factor of two results wastage of one file's worth of space on the device.